jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/CrossContextServletContext.java

```diff
@@ -113,7 +113,19 @@ public Set<String> getResourcePaths(String path)
     @Override
     public URL getResource(String path) throws MalformedURLException
     {
-        return _targetContext.getBaseResource().resolve(path).getURI().toURL();
+        try
+        {
+            return _targetContext.getBaseResource().resolve(path).getURI().toURL();
+        }
+        catch (MalformedURLException e)
+        {
+            throw e;
+        }
+        catch (Throwable e)
+        {
+            // catch IOException, RuntimeException, and things like java.nio.fileInvalidPathException here.
+            throw new MalformedURLException(path);
```

Comment:
We don't want to lose the original exception, so you either need to do:
```suggestion
            throw new MalformedURLException(path)
            {
                { 
                    initCause(cause);
                }
            };
```

or create a utility method/class somewhere that does that

Reply:
This was an intentional decision to not include the cause.  Mainly to avoid revealing FileSystem details on error pages.

Reply:
I added a new `org.eclipse.jetty.util.EnhancedMalformedURLException`. (perhaps it should be in jetty-io?)

If we have a need to not include specific causes (like we did in the past), we can add the logic there.